server/application/application.go

```diff
@@ -517,41 +513,85 @@ func (s *Server) deploy(
 		targetObjs[i] = obj
 	}
 
-	liveObjs, err := kube.GetLiveResources(config, targetObjs, namespace)
+	server, namespace := argoutil.ResolveServerNamespace(app.Spec.Destination, manifestInfo)
+
+	comparison, err := s.appComparator.CompareAppState(server, namespace, targetObjs, app)
 	if err != nil {
 		return nil, nil, err
 	}
-	diffResList, err := diff.DiffArray(targetObjs, liveObjs)
+
+	clst, err := s.clusterService.Get(ctx, &cluster.ClusterQuery{Server: server})
 	if err != nil {
 		return nil, nil, err
 	}
+	config := clst.RESTConfig()
 
 	var syncRes ApplicationSyncResult
 	syncRes.Resources = make([]*ResourceDetails, 0)
-	for i, diffRes := range diffResList.Diffs {
+	for _, resourceState := range comparison.Resources {
+		var liveObj, targetObj *unstructured.Unstructured
+
+		if resourceState.LiveState != "null" {
+			liveObj = &unstructured.Unstructured{}
+			err = json.Unmarshal([]byte(resourceState.LiveState), liveObj)
+			if err != nil {
+				return nil, nil, err
+			}
+		}
+
+		if resourceState.TargetState != "null" {
+			targetObj = &unstructured.Unstructured{}
+			err = json.Unmarshal([]byte(resourceState.TargetState), targetObj)
+			if err != nil {
+				return nil, nil, err
+			}
+		}
+
+		needsCreate := liveObj == nil
+		needsDelete := targetObj == nil
+
+		obj := targetObj
+		if obj == nil {
+			obj = liveObj
+		}
 		resDetails := ResourceDetails{
-			Name:      targetObjs[i].GetName(),
-			Kind:      targetObjs[i].GetKind(),
+			Name:      obj.GetName(),
+			Kind:      obj.GetKind(),
 			Namespace: namespace,
 		}
-		needsCreate := bool(liveObjs[i] == nil)
-		if !diffRes.Modified {
+
+		if resourceState.Status == appv1.ComparisonStatusSynced {
 			resDetails.Message = fmt.Sprintf("already synced")
 		} else if dryRun {
 			if needsCreate {
 				resDetails.Message = fmt.Sprintf("will create")
+			} else if needsDelete {
+				resDetails.Message = fmt.Sprintf("will delete")
```

Comment:
This needs to take consideration the prune flag. Can we use the term "ignored" if we decide not to delete the resource?

Reply:
done